diff --git a/dummy_hcd/dummy_hcd.c b/dummy_hcd/dummy_hcd.c
index f629563..cc2aec6 100644
--- a/dummy_hcd/dummy_hcd.c
+++ b/dummy_hcd/dummy_hcd.c
@@ -948,15 +948,6 @@ static void dummy_udc_set_speed(struct usb_gadget *_gadget,
 	dummy_udc_update_ep0(dum);
 }
 
-static void dummy_udc_async_callbacks(struct usb_gadget *_gadget, bool enable)
-{
-	struct dummy	*dum = gadget_dev_to_dummy(&_gadget->dev);
-
-	spin_lock_irq(&dum->lock);
-	dum->ints_enabled = enable;
-	spin_unlock_irq(&dum->lock);
-}
-
 static int dummy_udc_start(struct usb_gadget *g,
 		struct usb_gadget_driver *driver);
 static int dummy_udc_stop(struct usb_gadget *g);
@@ -969,7 +960,6 @@ static const struct usb_gadget_ops dummy_ops = {
 	.udc_start	= dummy_udc_start,
 	.udc_stop	= dummy_udc_stop,
 	.udc_set_speed	= dummy_udc_set_speed,
-	.udc_async_callbacks = dummy_udc_async_callbacks,
 };
 
 /*-------------------------------------------------------------------------*/
@@ -1029,6 +1019,7 @@ static int dummy_udc_start(struct usb_gadget *g,
 	spin_lock_irq(&dum->lock);
 	dum->devstatus = 0;
 	dum->driver = driver;
+	dum->ints_enabled = 1;
 	spin_unlock_irq(&dum->lock);
 
 	return 0;
