diff --git a/raw_gadget/raw_gadget.c b/raw_gadget/raw_gadget.c
index 5dfe7ab..5eedee5 100644
--- a/raw_gadget/raw_gadget.c
+++ b/raw_gadget/raw_gadget.c
@@ -543,12 +543,12 @@ static int raw_ioctl_run(struct raw_dev *dev, unsigned long value)
 	dev->state = STATE_DEV_REGISTERING;
 	spin_unlock_irqrestore(&dev->lock, flags);
 
-	ret = usb_gadget_register_driver(&dev->driver);
+	ret = usb_gadget_probe_driver(&dev->driver);
 
 	spin_lock_irqsave(&dev->lock, flags);
 	if (ret) {
 		dev_err(dev->dev,
-			"fail, usb_gadget_register_driver returned %d\n", ret);
+			"fail, usb_gadget_probe_driver returned %d\n", ret);
 		dev->state = STATE_DEV_FAILED;
 		goto out_unlock;
 	}
