diff --git a/dummy_hcd/dummy_hcd.c b/dummy_hcd/dummy_hcd.c
index 4c9d1e4..d0248c5 100644
--- a/dummy_hcd/dummy_hcd.c
+++ b/dummy_hcd/dummy_hcd.c
@@ -618,7 +618,21 @@ static int dummy_enable(struct usb_ep *_ep,
 		_ep->name,
 		desc->bEndpointAddress & 0x0f,
 		(desc->bEndpointAddress & USB_DIR_IN) ? "in" : "out",
-		usb_ep_type_string(usb_endpoint_type(desc)),
+		({ char *val;
+		 switch (usb_endpoint_type(desc)) {
+		 case USB_ENDPOINT_XFER_BULK:
+			 val = "bulk";
+			 break;
+		 case USB_ENDPOINT_XFER_ISOC:
+			 val = "iso";
+			 break;
+		 case USB_ENDPOINT_XFER_INT:
+			 val = "intr";
+			 break;
+		 default:
+			 val = "ctrl";
+			 break;
+		 } val; }),
 		max, ep->stream_en ? "enabled" : "disabled");
 
 	/* at this point real hardware should be NAKing transfers
